src/CodeStyle/Core/CodeFixes/Host/Mef/CodeStyleHostLanguageServices.cs.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Composition.Hosting;
+using System.Linq;
+using System.Reflection;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    internal sealed partial class CodeStyleHostLanguageServices : HostLanguageServices
+    {
+        private sealed class MefHostExportProvider : IMefHostExportProvider
```

Comment:
Why do we need another implementation of IMefHostExportProvider rather than reusing one of the existing ones? I get the CreateAssemblies method has to exist somewhere, but it seems you could just take that assembly list and pass it to

https://github.com/dotnet/roslyn/blob/59f4ccd75818fc5e9b923565c045e79c61fc85c2/src/Workspaces/Core/Portable/Workspace/Host/Mef/MefHostServices.cs#L43

Reply:
I recall I tried that but something about internal types that I wasn't able to extract to the shared layer prevented me from going that route. Hence, I ended up cloning the relevant code. Let me try that route again so this time I can clearly state where my attempt fell over.